app/src/test/java/com/example/trainingintive/viewmodels/DogImageViewModelTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.trainingintive.TestSchedulersProvider
+import com.example.trainingintive.model.DogImageUrl
+import com.example.trainingintive.repository.DogImageRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Single
+import org.amshove.kluent.`should be equal to`
+import org.junit.Rule
+import org.junit.Test
+
+class DogImageViewModelTest {
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: DogImageViewModel
+    private val sampleImageUrl = "https:\\/\\/images.dog.ceo\\/breeds\\/bulldog-french\\/n02108915_9666.jpg"
+    private val sampleStatus = "success"
+    private val sampleDog = DogImageUrl(sampleImageUrl, sampleStatus)
+    private val repository = mockk<DogImageRepository>() {
+        every { getDogImageUrl() } returns Single.just(sampleDog)
+        every { insertIntoLocalDatabase(any()) } returns Completable.complete()
+    }
+    private val schedulersForTests = TestSchedulersProvider()
+
+    @Test
+    fun `after init viewmodel, livedata should contains dog imiage url`() {
+        viewModel = DogImageViewModel(repository, schedulersForTests)
+
+        val observer = Observer<String> { }
+        viewModel.dog.observeForever(observer)
+        val imageUrl: String = viewModel.dog.value ?: ""
```

Comment:
Type is not necessary in this case I thing :) Don't specify type when it is not needed